### PR TITLE
[TASK] Rework smaller parts to add specific notes

### PR DESCRIPTION
**Pull request by garvinhicking**
(This may use some :ref: links which are committed seperately)

### CODE REVIEW

File: Documentation/Appendix/HowToDeprecateThings.rst [hunk 1]
```diff
@@ -18,6 +18,12 @@ or change functionality in the TYPO3 Core, it has to be deprecated first.
     Upon deprecating or removing functionality the change must be added
     to the :ref:`Extension scanner <extension-scanner>` whenever possible.
 
+..  note::
+    It's not always possible to "deprecate" everything. In some cases,
```
Comment (sbuerk) [inline Documentation/Appendix/HowToDeprecateThings.rst:22]: missing empty line ? :thinking:

File: Documentation/Appendix/OSX/InstallGrunt.rst [hunk 2]
```diff
@@ -10,6 +10,11 @@
 GRUNT install on OSX
 ====================
 
+..  hint::
+
+   You may not need this at all, if you are using :ref:`runTests.sh <runtests_sh>`
```
Comment (sbuerk) [inline Documentation/Appendix/OSX/InstallGrunt.rst:15]: alignment - 3 spaces vs 4 spaces ?

File: Documentation/Appendix/OSX/InstallGrunt.rst [hunk 3]
```diff
@@ -10,6 +10,11 @@
...
+   with dockerized environment for building all assets.
+
 First off, make sure you installed **npm** like described :ref:`here<osx-npm>`.
 
 In your Terminal app navigate to the folder that you cloned the TYPO3 source into. Switch into the ``Build`` folder and run
```
Comment (sbuerk) [inline Documentation/Appendix/OSX/InstallGrunt.rst:20]: Unrelated to this pr, but are double ticks (start + end paired) okay ? In TYPO3 we usualy use single ticks (vs doctrine for example).

File: Documentation/Appendix/OSX/InstallNpm.rst [hunk 4]
```diff
@@ -10,6 +10,11 @@
 Npm install on OSX
 ===================
 
+..  hint::
+
+   You may not need this at all, if you are using :ref:`runTests.sh <runtests_sh>`
```
Comment (sbuerk) [inline Documentation/Appendix/OSX/InstallNpm.rst:15]: Alignment 3 vs 4 spaces

File: Documentation/CheatSheets/Git.rst [hunk 5]
```diff
@@ -132,6 +132,8 @@ This assumes, you have correctly configured your remote as described in
 
 .. note::
    Pushing to `refs/publish` is deprecated, we now push to `refs/for`.
```
Comment (sbuerk) [inline Documentation/CheatSheets/Git.rst:134]: missing empty line ?

**Review by sbuerk — CHANGES_REQUESTED** [review]

File: Documentation/Appendix/HowToDeprecateThings.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/Appendix/HowToDeprecateThings.rst:22]: There is not rule about this and all combinations work.

File: Documentation/Appendix/OSX/InstallGrunt.rst (hunk 3 above)
Comment (linawolf) [reply Documentation/Appendix/OSX/InstallGrunt.rst:20]: The only difference in our implementation is that you can use a single backtick within double backticks without escaping

File: Documentation/Appendix/OSX/InstallGrunt.rst (hunk 2 above)
Comment (garvinhicking) [reply Documentation/Appendix/OSX/InstallGrunt.rst:15]: This is generally something left to clean-up, I would prefer to do this in a seperate PR

**Review by linawolf — DISMISSED** [review]

File: Documentation/Appendix/HowToDeprecateThings.rst [hunk 6]
```diff
@@ -18,6 +18,12 @@ or change functionality in the TYPO3 Core, it has to be deprecated first.
...
+    hard breaking changes must be introduced. These need to be adressed
```
Comment (froemken) [inline Documentation/Appendix/HowToDeprecateThings.rst:23]:
```suggestion
    hard breaking changes must be introduced. These need to be addressed
```

File: Documentation/ReportingAnIssue/Index.rst [hunk 7]
```diff
@@ -49,6 +54,13 @@ something similar has already been reported.
 easy to find existing issues. Filters are located on the the left hand
 side of the navigation menu. You can use this feature to help refine searches.
 
+Of course you are also able to use the search functionality of
+`Forge <https://forge.typo3.org>`__ itself, which offers some more specific
+filter options.
+
+You help the people maintaining the issue tracker a lot by first making sure
+your problem is not covered already. The less duplicate issues our team needs
+to triage, the more time we have for actually adressing bugs and features.
```
Comment (froemken) [inline Documentation/ReportingAnIssue/Index.rst:63]:
```suggestion
to triage, the more time we have for actually addressing bugs and features.
```

**Review by sbuerk — APPROVED** [review]

**Pull request merged**